src/Build.UnitTests/BackEnd/MSBuild_Tests.cs

```diff
@@ -772,6 +772,138 @@ public void ItemsIncludeExcludePathsCombinations()
             }
         }
 
+        /// <summary>
+        /// Referring to an item outside of target leads to 'naturally expected' reference to the item being processed.
+        ///  No expansion occurs.
+        /// </summary>
+        [Fact]
+        public void ItemsRecursionOutsideTarget()
+        {
+            string projectFile = null;
+
+            try
```

Comment:
You can probably make these a little cleaner with TestEnvironment. It does the cleanup for you automatically when it's disposed.